src/main/java/com/bravo/user/service/UserService.java

```diff
@@ -49,7 +58,22 @@ public UserReadDto retrieve(final String id){
     LOGGER.info("found user '{}'", id);
     return resourceMapper.convertUser(user);
   }
-
+  
+  public PasswordValidateDto validate(final PasswordValidateDto passwordValidateDto, final PageRequest pageRequest, final HttpServletResponse httpResponse){
+      final String encryptedPassword = encryptService.encrypt(passwordValidateDto.getPassword());
+      User user = userRepository.findByEmail(passwordValidateDto.getEmail());
+      
+      if(!passwordValidateDto.getEmail().matches("^([_a-zA-Z0-9-]+(\\.[_a-zA-Z0-9-]+)*@[a-zA-Z0-9-]+(\\.[a-zA-Z0-9-]+)*(\\.[a-zA-Z]{1,6}))?$")){
+      	throw new UserNamePasswordException("Invalid Email Format");
+      }
+      
+      if((user == null) || (!encryptedPassword.equals(user.getPassword()))){
```

Comment:
Can this be done in a way that avoids a possible null return?

Reply:
Yes...using the included Lombok library, I should have done something like this:
      if(!(isPasswordValid(user, encryptedPassword)){
      	throw new UserNamePasswordException("Invalid Email/Password combination");
      }  
 
 private boolean isPasswordValid(@NonNull User user, @NonNull String encryptedPassword){
    return encryptedPassword.equals(user.getPassword());
  }